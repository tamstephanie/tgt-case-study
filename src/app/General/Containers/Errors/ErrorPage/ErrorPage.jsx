import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import {Button, Typography, withStyles} from "@material-ui/core";
import {ArrowForward} from "@material-ui/icons";

import AppPage from "app/General/Containers/AppPage/AppPage";
import {ErrorPageStyle} from "./ErrorPage.style";

class ErrorPage extends PureComponent {
    render() {
        let {classes} = this.props;
        return (
            <div className={classes.errorPage + " error-page"}>
                <AppPage title="Page Not Found">
                    <Typography>
                        Oops! The page you are looking for was not found. It may be moved or not available.
                    </Typography>
                    <Button variant="outlined" href="#" endIcon={<ArrowForward />} color="secondary">
                        Return home
                    </Button>
                </AppPage>
            </div>
        );
    }
}

ErrorPage.propTypes = {
    /**
     * JSON object of CSS styling, which is then applied to the corresponding component(s)
     * @type {JSS}
     * @optional
     */
    classes: PropTypes.object,
};

export default withStyles(ErrorPageStyle)(ErrorPage);