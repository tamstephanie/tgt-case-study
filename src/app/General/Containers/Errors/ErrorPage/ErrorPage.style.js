export const ErrorPageStyle = (theme) => ({
    errorPage: {
        "& .app-page": {
            "& .MuiCard-root": {
                backgroundColor: theme.palette.secondary.main,
                "& .MuiTypography-root": {
                    color: theme.palette.common.main
                }
            }
        },
        "& .error-page-content": {
            marginLeft: "auto",
            marginRight: "auto",
            textAlign: "center"
        },
        "& .MuiButton-outlined": {
            marginTop: "50px"
        }
    }
});