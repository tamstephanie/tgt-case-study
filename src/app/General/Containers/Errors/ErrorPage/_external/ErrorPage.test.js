import React from "react";
import {create} from "react-test-renderer";

import ErrorPage from "../ErrorPage";

describe("ErrorPage", () => {
    it("should match the snapshot", () => {
        let page = create(<ErrorPage />);
        expect(page).toMatchSnapshot();
    });
});