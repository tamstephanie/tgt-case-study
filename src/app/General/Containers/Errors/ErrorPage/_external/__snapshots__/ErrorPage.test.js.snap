// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match the snapshot 1`] = `
<div
  className="ErrorPage-errorPage-1 error-page"
>
  <div
    className="app-page"
  >
    <div
      className="MuiContainer-root AppPage-header-2 header MuiContainer-maxWidthXl"
    >
      <div
        className="app-title-card"
      >
        <div
          className="MuiPaper-root MuiCard-root AppPage-appTitle-3 app-title MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h1
              className="MuiTypography-root MuiTypography-h1"
            >
              <b>
                Page Not Found
              </b>
            </h1>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiContainer-root AppPage-content-4 content MuiContainer-maxWidthLg"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthMd"
      >
        <div
          className="MuiPaper-root error-page-content MuiPaper-elevation0"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Oops! The page you are looking for was not found. It may be moved or not available.
          </p>
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
            href="#"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              Return home
              <span
                className="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
